compiler/rustc_hir_analysis/src/hir_ty_lowering/mod.rs

```diff
@@ -1070,13 +1070,46 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                 )?
             }
             (
-                &ty::Param(_),
-                Res::SelfTyParam { trait_: param_did } | Res::Def(DefKind::TyParam, param_did),
+                ty::Param(_),
+                Res::SelfTyParam { trait_: param_def_id }
+                | Res::Def(DefKind::TyParam, param_def_id),
             ) => self.probe_single_ty_param_bound_for_assoc_ty(
-                param_did.expect_local(),
+                param_def_id.expect_local(),
                 assoc_ident,
                 span,
             )?,
+            // FIXME(fmease):
+            // Should we require the pre-lowered projectee (the HIR QSelf) to be a `Res::Def(DefKind::AssocTy, _)`?
+            // Rephrased, should we flat out reject `Identity<T::Assoc>::Assoc` even if `T::Assoc::Assoc` typeck's?
+            // For comparison, `Identity<T>::Assoc` gets rejected even if `T::Assoc` passes typeck.
```

Comment:
ah, so wrt to my first comment, this is about non-lazy type aliases?

I would expect that this would then break with lazy type aliases unless we explicitly normalize them here. Because of that I feel like we should restrict it based on the res?

Reply:
Yeah, I'm referring to eager type aliases here. Makes sense to restrict them to have parity with lazy type aliases +1.

As mentioned in the source code comment, that's gonna be bit of pain to impl because `Res` is actually `Err` in this scenario right now.

Reply:
Yeah, I'm referring to eager type aliases here. Makes sense to restrict them to have parity with lazy type aliases +1.

As mentioned in the source code comment, that's gonna be bit of pain to impl because `Res` is actually `Err` in this scenario right now.